src/pages/overview/document-generation-api/fragments.md

```diff
@@ -89,9 +94,84 @@ And the output document generated will look like:
 
 ![Output of name fragment in document](../images/name_output.png)
 
-## Limitations
+## Other supported constructs
+
+### Conditional Phrases
+Evaluate value of fragments based on the conditions.
+
+```json
+[
+  {
+    "addressDetails": "<br>{{streetDetails}}<br>{{localityDetails}}",
+    "streetDetails": "<span style=\"color: 0000FF;\">{{addressline1}}<br>{{addressline2}}<br>{{addressline3}}</span>",
+    "localityDetails": "<span style=\"color: 006400;\">{% conditional-section expr(country=\"India\") %}{{street}} {{postcode}} {{city}},<i>{{country}}</i> {% end-section %}{% conditional-section expr(country=\"USA\") %}{{street}}{{city}} {{state}} {{zip}},<i>{{country}}</i> {% end-section %}</span>"
+  },
+  {
+    "fullname": "<span style=\"color: ff0000;\">{{firstname}} {{lastnameStyled}}</span>",
+    "lastnameStyled": "<span STYLE=\"font-size:14mm\"><b><i>{{lastname}}</i></b></span>"
+  }
+]
+```
+In the above example, the `addressDetails` depends upon `streetDetails` and `localityDetails`. But format of the address vary from country to country.
+For a country like India, the format of `localityDetails` is
+`street`
+`postCode city`
+`country`. For a country like USA, the format of `localityDetails` is
+`street`
+`city, state zip`
+`country`.
+
+![Output of fragment with condition in document](../images/fragments-condition.png)
+
+
+### Optional
+Setting the tag as optional will replace the tag with empty string if it's value is not present in input data and fragment data.
+
+```json
+[
+  {
+    "fullname": "{{firstname}} {{middlename:optional(true)}} {{lastname}}"
+  },
+  {
+    ...
+  }
+]
+```
+
+In the above example, the fragment tag `fullName` which is the composition of other tags namely `firstName`, `middleName` and `lastName`. But there are some people who don't have `middleName`. So, middleName needs to be optional, i.e, it should be ignored if it's value doesn't exist.
+
+![Output of fragment with optional value in document](../images/fragments-optional.png)
+
+### Default
+A default value can be specified for the tag used inside the fragment. In case, the specified tag is not present in the input data or fragment data, that tag gets replaced with the default value.
+```json
+[
+  {
+    "team": "ABC, {{organisation:default-val(\"Org\")}}"
+  },
+  {
+    ...
+  }
+]
+```
+
+In the above example, if the tag `organisation` does not exist in fragment or normal json, then default vaule (i.e. "Org") will be used.
+
+![Output of fragment with default value in document](../images/fragments-default.png)
+
+### Prefix
+Prefix is value appended before that tag used inside the fragment.
+```json
+[
+  {
+    "doctor": "{{fullname:prefix(\"Dr. \")}}",
+    "fullname": "{{firstname}} {{middlename:optional(true)}} {{lastname}}"
+  },
+  {
+    ...
+  }
+]
+```
+
```

Comment:
In the example above, `doctor` fragment tag is composed of 'fullname' fragment tag. Prefix property is used with `fullname` fragment tag. Prefix value of **Dr. ** will be added before the value of the `fullname` fragment tag in the output.